src/cert_provider/main.cc

```diff
@@ -0,0 +1,154 @@
+#include <stdlib.h>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+#include "json/json.h"
+
+#include "bootstrap.h"
+#include "config.h"
+#include "crypto.h"
+#include "httpclient.h"
+#include "logger.h"
+#include "utils.h"
+
+/*****************************************************************************/
+
+namespace bpo = boost::program_options;
+
+void check_info_options(const bpo::options_description &description, const bpo::variables_map &vm) {
+  if (vm.count("help") != 0) {
+    std::cout << description << '\n';
+    exit(EXIT_SUCCESS);
+  }
+  if (vm.count("version") != 0) {
+    std::cout << "Current cert_provider version is: " << AKTUALIZR_VERSION << "\n";
+    exit(EXIT_SUCCESS);
+  }
+}
+
+bpo::variables_map parse_options(int argc, char *argv[]) {
+  bpo::options_description description("CommandLine Options");
+  // clang-format off
+  description.add_options()
+      ("help,h", "help screen")
+      ("version,v", "Current cert_provider version")
+      ("credentials,c", bpo::value<std::string>()->required(), "zipped credentials file")
+      ("target,t", bpo::value<std::string>(), "target device")
+      ("port,p", bpo::value<int>(), "target port");
+  // clang-format on
+
+  bpo::variables_map vm;
+  std::vector<std::string> unregistered_options;
+  try {
+    bpo::basic_parsed_options<char> parsed_options =
+        bpo::command_line_parser(argc, argv).options(description).allow_unregistered().run();
+    bpo::store(parsed_options, vm);
+    check_info_options(description, vm);
+    bpo::notify(vm);
+    unregistered_options = bpo::collect_unrecognized(parsed_options.options, bpo::include_positional);
+    if (vm.count("help") == 0 && !unregistered_options.empty()) {
+      std::cout << description << "\n";
+      exit(EXIT_FAILURE);
+    }
+  } catch (const bpo::required_option &ex) {
+    // print the error and append the default commandline option description
+    std::cout << ex.what() << std::endl << description;
+    exit(EXIT_SUCCESS);
+  } catch (const bpo::error &ex) {
+    check_info_options(description, vm);
+
+    // print the error message to the standard output too, as the user provided
+    // a non-supported commandline option
+    std::cout << ex.what() << '\n';
+
+    // set the returnValue, thereby ctest will recognize
+    // that something went wrong
+    exit(EXIT_FAILURE);
+  }
+
+  return vm;
+}
+
+/*****************************************************************************/
+int main(int argc, char *argv[]) {
+  loggerInit();
+  loggerSetSeverity(static_cast<LoggerLevels>(2));
+
+  boost::filesystem::remove("device_id");
+  boost::filesystem::remove("root.crt");
+  boost::filesystem::remove("client.pem");
+  boost::filesystem::remove("pkey.pem");
```

Comment:
What about using `TemporaryFile` from utils.h here?

Reply:
@cajun-rat I thought about that. I was trying to allow for a use case in which the user wanted to manually do something with the certificate files instead of having them automatically `scp`ed. If that use case is unlikely or undesirable, then yes, `TemporaryFile` would be wise. Is that your preference?

Reply:
Manually copying the files to somewhere is a reasonable use case.  I'm not sure about creating (or deleting) files with magic names in the current working directory though.

Maybe add a second option to dump the files in a directory instead/as well as copying them via scp?